component/azstorage/file_share.go

```diff
@@ -355,6 +415,7 @@ func (fs *FileShare) RenameDirectory(source string, target string) error {
 			err = fs.RenameDirectory(filepath.Join(source, dirInfo.Name), filepath.Join(target, dirInfo.Name))
 			if err != nil {
 				log.Err("FileShare::RenameDirectory : Failed to move subdirectories to new directory  %s", err.Error())
+				fs.DeleteDirectory(target)
```

Comment:
If rename fails, why we want to delete targer directory. It might be the case where targer already exists and this move command is more like moving some more files in that directory.

Reply:
I added those lines to delete the target directory because one of the unit tests failed without it. The test wanted to rename a nonexistent src directory to some target dir. Since the src dir didn't exist, it would error out and keep the target directory, which caused an assertion to fail. But you made a good point; I'll add an if statement to catch this edge case where the target dir already exists.

Reply:
if source directory does not exists then why rename will create the destination directory, sorry but dint get the context here.

Reply:
If the source directory doesn't exist, the rename will create the destination directory, because in the current code for this method, the second line in the method (after the first line of logging) is to create the target directory. I put it there because it's a recursive method, so it's handy for the subdirectories. 
But to address the cases where the source doesn't exist and the target already exists, I'll put an if statement before calling CreateDirectory() so that it won't create the destination directory unnecessarily (and delete the DeleteDirectory() calls). The if-statement would check if GetAttr() on the source directory has no errors to validate whether it's a valid directory or not.

Reply:
I meant GetProperties() rather than GetAttr()